### DISCUSSION TITLE
InMemoryDataset new input parameter

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#5558] [Q&A] [answered]
Can someone help define the init class shown here. I want to add a parameter called `split`, where its a float value like 0.25 that would be used later in the `process` section of the data.

```
class ImbCircuitDataset(InMemoryDataset):
    def __init__(self, root, transform=None, pre_transform=None,
                 pre_filter=None):
        super().__init__(root, transform, pre_transform, pre_filter)
        self.data, self.slices = torch.load(self.processed_paths[0])
```

so when I run `dataset = ImbCircuitDataset('test')`, i want to run `dataset = ImbCircuitDataset(split=0.25, 'test')` so i can automate my splitting process.

**Comment by EdisonLeeeee** [comment, marked as answer]
Did you mean that?
```python
class ImbCircuitDataset(InMemoryDataset):
    def __init__(self, root, split, transform=None, pre_transform=None,
                 pre_filter=None):
        super().__init__(root, transform, pre_transform, pre_filter)
        self.split = split
        self.data, self.slices = torch.load(self.processed_paths[0])
```

**Reply by anthonysirico** [reply]
The returns `AttributeError: 'ImbCircuitDataset' object has no attribute 'split'`

**Reply by EdisonLeeeee** [reply]
Oh my bad. `self.split` should be defined before `super().__init__`, similar to https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/datasets/planetoid.py

**Reply by anthonysirico** [reply]
Thanks, friend. That worked!
